src/components/Timeline.tsx

```diff
@@ -50,13 +59,10 @@ export default observer(function Timeline() {
           position="relative"
           height={10}
           bgColor="gray.500"
-          onClick={action((e) => {
-            if (rulerBoxRef.current)
-              timer.setTime(
-                uiStore.xToTime(
-                  e.clientX - rulerBoxRef.current.getBoundingClientRect().x,
-                ),
-              );
+          onMouseDown={action((e) => {
+            rulerDrag(e.nativeEvent);
+            window.addEventListener('mousemove', rulerDrag);
+            window.addEventListener('mouseup', ()=>window.removeEventListener('mousemove', rulerDrag));
```

Comment:
Nice! We can also remove this last event listener by giving `{ once: true }` to addEventListener as the second parameter (https://stackoverflow.com/a/70025020(

Reply:
And yeah, the more reacty way would be more verbose AFAICT, so this seems good to me